### PR TITLE
Introduce references

### PR DESCRIPTION
**Pull request by bierik**
Fixes https://github.com/4teamwork/ftw.showroom/issues/23

Provide possibility to open showroom items with links.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -245,6 +245,32 @@ This function can be overridden, see Configuration_.
     }
   );
 
+References
+----------
+
+References are useful if you want to open a showroom item with some additional links out of a text or something.
```
Comment (deiferni) [inline README.rst]: with some additional links out of a text or something -> that is referenced by more than one element on the same page.

File: README.rst [hunk 2]
```diff
@@ -245,6 +245,32 @@ This function can be overridden, see Configuration_.
...
+You just have to add the **showroom-reference** class and the **data-showroom-target-item** attribute to make the connection.
+The **data-showroom-target-item** attribute contains an id which references a showroom item on the page.
+You have to set the **data-showroom-id** by yourself on the showroom item to make the connection with the reference.
```
Comment (deiferni) [inline README.rst]: by yourself -> manually

File: README.rst [hunk 3]
```diff
@@ -245,6 +245,32 @@ This function can be overridden, see Configuration_.
...
+But you have to call `showroom.refresh()` to refresh the references.
+
+.. code:: html
+
+  <!-- The showroom item -->
+
+  <a href="#"
+    class="showroom-item"
+    data-showroom-id="your-unique-id"
+    data-showroom-target="http://www.google.ch"
```
Comment (deiferni) [inline README.rst]: i'd use `example.com` as example domains

File: ftw/showroom/js/src/showroom.js [hunk 4]
```diff
@@ -173,6 +182,11 @@ module.exports = function Showroom(items = [], options) {
     checkArrows();
   }
 
+  function refresh() {
+    options.references = Array.prototype.slice.call(document.querySelectorAll(".showroom-reference"));
```
Comment (deiferni) [inline ftw/showroom/js/src/showroom.js]: Should we make this class configurable?

File: ftw/showroom/resources/showroom.js [hunk 5]
```diff
@@ -432,7 +431,7 @@ module.exports = function Showroom() {
   function select(event) {
     event.preventDefault();
     var item = register.items.filter(function (item) {
-      return item.id === event.currentTarget.getAttribute("data-showroom-id");
+      return item.id === (event.currentTarget.getAttribute("data-showroom-id") || event.currentTarget.getAttribute("data-showroom-target-item"));
```
Comment (deiferni) [inline ftw/showroom/resources/showroom.js]: if i understand correctly `data-showroom-id` and `data-showroom-target-item` are mutually exclusive, maybe we should enforce this somewhere?

File: ftw/showroom/js/src/showroom.js (hunk 4 above)
Comment (bierik) [inline ftw/showroom/js/src/showroom.js]: In which case would you have an other class? This is just a convention. I think it doesn't matter which class we use.

Comment (deiferni) [inline ftw/showroom/js/src/showroom.js]: i thought the `showroom-item` class is already defined by the client, so for the sake of consistency we could provide a configuration option. but you're right it is optional 😉 .

**Comment by bierik** [comment]
@deiferni updated with your suggestions

**Comment by bierik** [comment]
@deiferni rebased

**Pull request merged**